### PR TITLE
Update protoc and regenerate common protos

### PR DESCRIPTION
**Pull request by jskeet**
Note that this doesn't yet update the version of gRPC tools that we're using, but we can do that later.

Note that a couple of proto messages are deleted as part of this, which is fine as we'll be releasing a 2.0.0-beta01 when we're ready.
This is part of addressing #368.

### CODE REVIEW

File: Google.Api.CommonProtos/Type/DateExtensions.cs [hunk 1]
```diff
@@ -15,11 +15,11 @@ namespace Google.Type
     public static class DateExtensions
     {
         /// <summary>
-        /// Converts the <see cref="DateTime.Date"/> part of <see cref="DateTime"/> to <see cref="Date"/>.
+        /// Converts the <see cref="System.DateTime.Date"/> part of <see cref="System.DateTime"/> to <see cref="Date"/>.
         /// </summary>
         /// <param name="dateTime">The <see cref="DateTime"/> instance being converted.</param>
```
Comment (chrisdunelm) [inline Google.Api.CommonProtos/Type/DateExtensions.cs]: Does this `see` reference also need to be `System.DateTime`?

File: Google.Api.CommonProtos/Type/DatePartial.cs [hunk 2]
```diff
@@ -14,13 +14,13 @@ public partial class Date
         /// <summary>
         /// Converts <see cref="Date"/> to <see cref="DateTime"/>.
         /// </summary>
-        /// <returns>The converted <see cref="DateTime"/> with time at midnight and <see cref="DateTime.Kind"/> of <see cref="DateTimeKind.Unspecified"/>.</returns>
+        /// <returns>The converted <see cref="DateTime"/> with time at midnight and <see cref="System.DateTime.Kind"/> of <see cref="DateTimeKind.Unspecified"/>.</returns>
```
Comment (chrisdunelm) [inline Google.Api.CommonProtos/Type/DatePartial.cs]: `System.DateTime` in the first `see` here too?

File: Google.Api.CommonProtos/Type/DatePartial.cs [hunk 3]
```diff
@@ -14,13 +14,13 @@ public partial class Date
         /// <summary>
         /// Converts <see cref="Date"/> to <see cref="DateTime"/>.
```
Comment (chrisdunelm) [inline Google.Api.CommonProtos/Type/DatePartial.cs]: `System.DateTime`?

**Comment by jskeet** [comment]
Well spotted - thanks, fixed those three.

**Pull request merged**